share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,628 @@
+## Wake Tutorial
+
+### Invoking wake
+
+1. Create a wake database: `wake --init .`
+
+2. Run command for wake: `wake -vx '<function_name> <parameter_if_any>'`
+
+* If you want to display a string you can that using the below function by writing the string to be displayed inisde "" and call the function abc . Ex: wake -x 'abc'
+```
+def abc = "Hello World"
+```
+
+* The other way to do this is `wake -x '"Hello World"'`
+
+### Defining Functions
+A function may be defines using the keyword def. Simple function definition takes the form
+`def <fun name> <parameter> = <expression>`
+
+* Below function  double takes an argment x and gives the output by multiplying the x by 2
+```
+def double x = 2*x
+```
+
+* Below function triple takes an argument x and gives the output by multiplying the x by 3
+```
+def triple x = 3*x
+```
+
+* Function inc increments the values by 1 and returns the value
+```
+def inc x = x+1
+```
+
+* Function adda takes an argument of type string and concantenates it with b and gives the output
+```
+def adda s = "{s}b"
+```
+
+* Function times4 takes an argument x and applies the function double twice on x
+```
+def times4 x = double (double x)
+```
+
+* Function aveI takes 2 arguments x and y and adds two values and divide the result by 2. This function shows the other was of defining the parameters using lambda operator
+```
+def aveI = \x\y (x+y)/2
+```
+
+* Function aveR takes 2 arguments of type double and divides the result by 2
+```
+def aveR = \x\y (x +. y) /. 2.0
+```
+
+* Function aveR adds 2 values and provides the remainder 
+```
+def aveM = \x\y (x+y) % 2
+```
+
+* Function duplicate concantenates or copies a given string twice
+```
+def duplicate s  ="{s}{s}"
+```
+
+* integerToUnicode function takes the ASCII input and provides its equivalent character
+```
+wake -x 'integerToUnicode 99'
+```
+
+* unicodeToInteger function gives the ASCII value of the given character
+```
+wake -x 'unicodeToInteger "a"'
+```
+
+* strlen function returns the length of the given string
+```
+wake -x 'strlen "abcde"'
+```
+
+* Below function takes a string and returns the middle character of the string
+```
+def middle fullstring =
+  def len = if (strlen fullstring % 2 == 0) then (strlen fullstring / 2) else  ((strlen fullstring / 2) + 1)
```

Comment:
now we are showing `if`, `then`, `explode`, `take`, `reverse`, `head`... none of which you have explained. None of this is making logical sense under "Defining Functions" heading.

Reply:
I will explain the built-in functions in the next commit. This is more of playing around the strings and extracting different outputs from them